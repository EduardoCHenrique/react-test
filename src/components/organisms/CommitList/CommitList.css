@import '../../../style/index.css';

.commit-list {
  height: calc(100vh - var(--header-height));
  overflow: scroll;
}
