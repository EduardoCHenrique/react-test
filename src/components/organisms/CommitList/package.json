{
  "name": "CommitList",
  "private": true,
  "main": "./CommitList.js"
}
