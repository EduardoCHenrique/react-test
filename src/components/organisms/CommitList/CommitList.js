import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import Loader from 'src/components/atoms/Loader';
import Commit from 'src/components/molecules/Commit';
import InfiniteScroll from 'react-infinite-scroller';

import './CommitList.css';

const renderCommit = (commit, index) => <Commit key={`commit-${index}`} {...commit} />

const CommitList = ({className, commits, handleLoadMoreCommits}) => {
  return (
    <div className={classNames('commit-list container', className)}>
      <InfiniteScroll
        loadMore={handleLoadMoreCommits}
        hasMore={true}
        loader={<Loader />}
        useWindow={false}
        initialLoad={false}>
        {commits.map(renderCommit)}
      </InfiniteScroll>
    </div>
  )
}

CommitList.propTypes = {
  className: PropTypes.string,
  commits: PropTypes.array,
  fetchingCommits: PropTypes.bool,
  commitsError: PropTypes.bool
}

export default CommitList