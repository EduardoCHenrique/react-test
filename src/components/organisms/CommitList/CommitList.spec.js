import React from 'react';
import { shallow } from 'enzyme';
import CommitList from './CommitList.js';
import faker from 'faker'

const default_props = {
  className: faker.random.word(),
  commits: [],
  fetchingCommits: false,
  commitsError: false
};

function setup(props = default_props) {
  const wrapper = shallow(<CommitList {...props} />);
  return { wrapper, props };
}

describe('CommitList Test Suite', () => {
  it('Should match snapshot', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});