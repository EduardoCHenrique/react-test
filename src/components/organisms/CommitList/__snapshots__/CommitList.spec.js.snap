// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommitList Test Suite Should match snapshot 1`] = `ShallowWrapper {}`;
